plugins/inputs/x509_cert/x509_cert_test.go

```diff
@@ -463,3 +470,197 @@ func TestServerName(t *testing.T) {
 		})
 	}
 }
+
+// Bases on code from
+// https://medium.com/@shaneutt/create-sign-x509-certificates-in-golang-8ac4ae49f903
+func TestClassification(t *testing.T) {
+	start := time.Now()
+	end := time.Now().AddDate(0, 0, 1)
+	tmpDir, err := os.MkdirTemp("", "telegraf-x509-*")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	// Create the CA certificate
+	caPriv, err := rsa.GenerateKey(rand.Reader, 4096)
+	require.NoError(t, err)
+
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(342350),
+		Subject: pkix.Name{
+			Organization: []string{"Testing Inc."},
+			Country:      []string{"US"},
+			CommonName:   "Root CA",
+		},
+		NotBefore:             start,
+		NotAfter:              end,
+		IsCA:                  true,
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPriv.PublicKey, caPriv)
+	require.NoError(t, err)
+	caPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: caBytes})
+
+	// Write CA cert
+	f, err := os.Create(filepath.Join(tmpDir, "ca.pem"))
```

Comment:
`path/filepath` is [OS aware](https://pkg.go.dev/path/filepath#Join) do we need to use [`path`](https://pkg.go.dev/path#Join) instead only for testing?

Reply:
The idea is to use a OS-aware URI. Fixed the test by fixing the expectations (hopefully).